File: backend/app/Modules/Secretariat/Models/LodgeSession.php
<?php

namespace App\Modules\Secretariat\Models;

use App\Modules\Lodges\Models\Lodge;
use Illuminate\Database\Eloquent\Factories\HasFactory;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Database\Eloquent\Relations\HasOne;
use Illuminate\Database\Eloquent\SoftDeletes;

class LodgeSession extends Model
{
    use HasFactory, SoftDeletes;

    protected $fillable = [
        'lodge_id',
        'date',
        'type',
        'degree',
        'rite',
        'tronco',
        'summary',
    ];

    protected function casts(): array
    {
        return [
            'date' => 'date',
            'tronco' => 'decimal:2',
        ];
    }

    public function lodge(): BelongsTo
    {
        return $this->belongsTo(Lodge::class);
    }

    public function balaustre(): HasOne
    {
        return $this->hasOne(Balaustre::class);
    }
}

File: mobile/lib/models/lodge_session.dart
class LodgeSession {
  final int id;
  final String date;
  final String type;
  final String degree;
  final String rite;
  final String? summary;

  LodgeSession({
    required this.id,
    required this.date,
    required this.type,
    required this.degree,
    required this.rite,
    this.summary,
  });

  factory LodgeSession.fromJson(Map<String, dynamic> json) {
    return LodgeSession(
      id: json['id'] as int,
      date: json['date'] as String,
      type: json['type'] as String,
      degree: json['degree'] as String,
      rite: json['rite'] as String,
      summary: json['summary'] as String?,
    );
  }
}
